livre/eq/render/RayCastRenderer.cpp

```diff
@@ -538,8 +545,37 @@ struct RayCastRenderer::Impl
         glUseProgram( 0 );
     }
 
-    void onFrameEnd()
+    void renderAxis( const Frustum& frustum )
+    {
+        GLSLShaders::Handle program = _axisShaders.getProgram( );
+        LBASSERT( program );
+        glUseProgram( program );
+
+        GLint tParamNameGL = glGetUniformLocation( program, "renderTexture" );
+        glUniform1i( tParamNameGL, 0 );
+
+        tParamNameGL = glGetUniformLocation( program, "modelViewProjectionMatrix" );
+        glUniformMatrix4fv( tParamNameGL, 1, false, frustum.getMVPMatrix( ).array );
+
+        tParamNameGL = glGetUniformLocation( program, "modelView" );
+        glUniformMatrix4fv( tParamNameGL, 1, false, frustum.getMVMatrix().array );
+
+        tParamNameGL = glGetUniformLocation( program, "normalMatrix" );
+        glUniformMatrix4fv( tParamNameGL, 1, false, frustum.getNormalMatrix( ).array );
+
+        tParamNameGL = glGetUniformLocation( program, "invProjectionMatrix" );
+        glUniformMatrix4fv( tParamNameGL, 1, false, frustum.getInvProjMatrix( ).array );
```

Comment:
Yes I nedd it in the shader.
"vec3 pixelWorldSpacePos = vec3(( invModelViewMatrix * pixelEyeSpacePos ).xyz );"